content/python/vlanTagging.md

```diff
@@ -1,18 +1,164 @@
 ---
-title: "Tag Vlans"
-description: "Some examles of how to add vlans to a server. AKA Vlan Trunks."
-date: "2017-06-09"
+title: "Trunking VLANs"
+description: "Some examles of how to adding VLAN Trunks. AKA VLAN Tagging"
+date: "2022-08-04"
 classes: 
     - "SoftLayer_Network_Component"
 tags:
     - "vlan"
     - "server"
 ---
+This example shows how to add VLAN Trunks to a server.
 
-A rough examples of how to add vlans to servers
+*NOTE* When trying to get the vlan tags/trunks from a server, the trunk information is on the `Hardware_Server->networkComponent->uplinkComponent->vlanTrunks` relationship. Esentially the Server's NIC doesn't know about network VLANs, but the switch port that is connected to the NIC does.
 
 
-*NOTE* When trying to get the vlan tags/trunks from a server, the trunk information is on the `Hardware_Server->networkComponent->uplinkComponent->vlanTrunks` relationship. 
+## VLAN Trunks
+
+```python
+"""
+@author Christopher Gallo
+
+Adds specified Public and Private VLANs to a given server ID.
+Script requried
+Python 3.7+
+click 8.0+
+rich 
+"""
+
+# Click is used to make this script support CLI arguments
+import click
+# Used for details table output
+from rich.console import Console
+from rich.table import Table
+
+import SoftLayer
+from pprint import pprint as pp 
+
+class testVlanTrunk():
+    def __init__(self):
+        """SoftLayer Client"""
+        self.client = SoftLayer.Client()
+        debugger = SoftLayer.DebugTransport(self.client.transport)
+        self.client.transport = debugger
+
+    def trunkPublic(self, serverId, vlans):
+        """Adds Trunks to a Public network component"""
+        component = self.getNetworkComponents(serverId, private=False)
+        networkVlans = self.listToVlan(vlans)
+        self.addTrunks(component.get('id'), networkVlans)
+       
+    def trunkPrivate(self, serverId, vlans):
+        """Adds Trunks to a Private network component"""
+        component = self.getNetworkComponents(serverId, private=True)
+        networkVlans = self.listToVlan(vlans)
+        self.addTrunks(component.get('id'), networkVlans)
```

Comment:
The "type" argument is not supplied here and seems that "Private" was the intent.